### DISCUSSION TITLE
Calling Rest api and set output as env variable

### DISCUSSION THREAD

**Discussion by pradeepkusingh** [concourse/concourse#6424] [Help & Support]
I have a requirement where I can't store secrets in vault as it gets frequently changed, and takes lot of cycles to update in vault. but these secrets can be accessed using a rest api, so I would like to access rest api and use output as env variable to use across jobs/pipeline. What is the best way to achieve it?
Any help would be highly appreciated.

**Comment by taylorsilva** [comment]
I think you can use a task + `load_var` step to accomplish this, if you're okay with writing some of your own code for the task step.

The task would make the API call and save the value of the var to a file which will be the output of the task.
The next step would then be a `load_var` step: https://concourse-ci.org/jobs.html#load-var-step
The `load_var` step would read the contest of the output file from the task step and make it available for future steps to load as a var `((my-var))`.
